backend/test/integration/experiment_api_test.go

```diff
@@ -40,6 +40,12 @@ func (s *ExperimentApiTest) SetupTest() {
 	}
 }
 
+func (s *ExperimentApiTest) TearDownTest() {
```

Comment:
I have a high level comment: 

Would it be possible to run all the tests on both: 
- A non upgraded cluster. 
- A cluster that is upgraded (DB for now, processes when we get there). 

I am a bit worried that we must create special tests for the upgrade case. It seems that we should just reuse the same tests, otherwise there are many things that we don't test.  

Could you also add comments in the code (TODO) that explain how we will test upgrades from different past versions?